units/scheduler.go

```diff
@@ -40,20 +43,28 @@ func makeDistroSchedulerJob() *distroSchedulerJob {
 	return j
 }
 
-func NewDistroSchedulerJob(distroID string, ts time.Time) amboy.Job {
+func NewDistroSchedulerJob(env evergreen.Environment, distroID string, ts time.Time) amboy.Job {
 	j := makeDistroSchedulerJob()
 	j.DistroID = distroID
 	j.SetID(fmt.Sprintf("%s.%s.%s", schedulerJobName, distroID, ts.Format(tsFormat)))
 
+	j.env = env
 	return j
 }
 
 func (j *distroSchedulerJob) Run(ctx context.Context) {
 	defer j.MarkComplete()
 
+	if j.env == nil {
+		j.env = evergreen.GetEnvironment()
+	}
+
+	settings := j.env.Settings()
```

Comment:
Should this look up the settings from the db instead? That way you can change the allocator on the fly without having to restart the service